src/evm/check-conditions.ts

```diff
@@ -43,20 +44,27 @@ export const checkSubscriptionCondition = async (
         .connect(voidSigner)
         .functions[conditionFunction.name](subscription.id, encodedFulfillmentData, conditionParameters),
     {
-      timeoutMs: DEFAULT_RETRY_TIMEOUT_MS,
+      timeoutMs: TIMEOUT_MS,
+      retries: 1,
```

Comment:
This can be also a constant in `constants.ts`.
I'm not sure how did the previous `retryGo` implementation look like, was it just one retry?

Reply:
The previous **retryGo** was:

```ts
export const retryGo = <T>(fn: () => Promise<T>, options?: node.utils.PromiseOptions) =>
  node.utils.go(() => node.utils.retryOnTimeout(DEFAULT_RETRY_TIMEOUT_MS, fn), options);
```
and **retryOnTimeout** was:

```ts
export function retryOnTimeout<T>(maxTimeoutMs: number, operation: () => Promise<T>, options?: ContinuousRetryOptions) {
  const promise = new Promise<T>((resolve, reject) => {
    function run(): Promise<any> {
      // If the promise is successful, resolve it and bubble the result up
      return operation()
        .then(resolve)
        .catch((reason: any) => {
          // Only if the error is a timeout error, do we retry the promise
          if (reason instanceof Error && reason.message.includes('Operation timed out')) {
            // Delay the new attempt slightly
            return sleep(options?.delay || DEFAULT_RETRY_DELAY_MS)
              .then(run)
              .then(resolve)
              .catch(reject);
          }

          // If the error is NOT a timeout error, then we reject immediately
          return reject(reason);
        });
    }

    return run();
  });

  return promiseTimeout(maxTimeoutMs, promise);
}
```